make_image: ensure correct quoting of variables when calling genisoimage
    
By first assembling the command in a variable and then using eval to execute it
we avoid both duplication and errors passing quoted options in $MKISOFS_OPTS.

diff --git a/tools/make_image b/tools/make_image
index 85732c7..19decce 100755
--- a/tools/make_image
+++ b/tools/make_image
@@ -85,63 +85,39 @@ do
 	# here... :-(
     if [ $JIGDO_WANTED = 0 ] && [ $ISO_WANTED = 1 ] ; then
 
-        echo $MKISOFS $MKISOFS_OPTS -V "$volid" -o $OUT/$OUTFILE.iso $opts $dirs CD$n
-        echo $MKISOFS $MKISOFS_OPTS -V "$volid" -o $OUT/$OUTFILE.iso $opts $dirs CD$n > CD$n/.disk/mkisofs
-        $MKISOFS $MKISOFS_OPTS -V "$volid" -o $OUT/$OUTFILE.iso $opts $dirs CD$n
+        CMD="$MKISOFS $MKISOFS_OPTS -V '$volid' \
+            -o $OUT/$OUTFILE.iso $opts $dirs CD$n"
+        echo $CMD
+        echo $CMD > CD$n/.disk/mkisofs
+        eval "$CMD"
 
     elif [ $JIGDO_WANTED = 1 ] && [ $ISO_WANTED = 1 ] ; then
 
-        echo $MKISOFS $MKISOFS_OPTS -V "$volid" \
+        CMD="$MKISOFS $MKISOFS_OPTS -V '$volid' \
             -o $OUT/$OUTFILE.iso \
             -jigdo-jigdo $OUT/$OUTFILE.jigdo \
             -jigdo-template $OUT/$OUTFILE.template \
             -jigdo-map Debian=$MIRROR/ \
             -jigdo-exclude boot$n \
             -md5-list $DIR/md5-check \
-            $JIGDO_OPTS $opts $dirs CD$n
-        echo $MKISOFS $MKISOFS_OPTS -V "$volid" \
-            -o $OUT/$OUTFILE.iso \
-            -jigdo-jigdo $OUT/$OUTFILE.jigdo \
-            -jigdo-template $OUT/$OUTFILE.template \
-            -jigdo-map Debian=$MIRROR/ \
-            -jigdo-exclude boot$n \
-            -md5-list $DIR/md5-check \
-            $JIGDO_OPTS $opts $dirs CD$n > CD$n/.disk/mkisofs
-        $MKISOFS $MKISOFS_OPTS -V "$volid" \
-            -o $OUT/$OUTFILE.iso \
-            -jigdo-jigdo $OUT/$OUTFILE.jigdo \
-            -jigdo-template $OUT/$OUTFILE.template \
-            -jigdo-map Debian=$MIRROR/ \
-            -jigdo-exclude boot$n \
-            -md5-list $DIR/md5-check \
-            $JIGDO_OPTS $opts $dirs CD$n
+            $JIGDO_OPTS $opts $dirs CD$n"
+        echo $CMD
+        echo $CMD > CD$n/.disk/mkisofs
+        eval "$CMD"
 
     elif [ $JIGDO_WANTED = 1 ] && [ $ISO_WANTED = 0 ] ; then
 
-        echo $MKISOFS $MKISOFS_OPTS -V "$volid" \
-            -o /dev/null \
-            -jigdo-jigdo $OUT/$OUTFILE.jigdo \
-            -jigdo-template $OUT/$OUTFILE.template \
-            -jigdo-map Debian=$MIRROR/ \
-            -jigdo-exclude boot$n \
-            -md5-list $DIR/md5-check \
-            $JIGDO_OPTS $opts $dirs CD$n
-        echo $MKISOFS $MKISOFS_OPTS -V "$volid" \
-            -o /dev/null \
-            -jigdo-jigdo $OUT/$OUTFILE.jigdo \
-            -jigdo-template $OUT/$OUTFILE.template \
-            -jigdo-map Debian=$MIRROR/ \
-            -jigdo-exclude boot$n \
-            -md5-list $DIR/md5-check \
-            $JIGDO_OPTS $opts $dirs CD$n > CD$n/.disk/mkisofs
-        $MKISOFS $MKISOFS_OPTS -V "$volid" \
+        CMD="$MKISOFS $MKISOFS_OPTS -V '$volid' \
             -o /dev/null \
             -jigdo-jigdo $OUT/$OUTFILE.jigdo \
             -jigdo-template $OUT/$OUTFILE.template \
             -jigdo-map Debian=$MIRROR/ \
             -jigdo-exclude boot$n \
             -md5-list $DIR/md5-check \
-            $JIGDO_OPTS $opts $dirs CD$n
+            $JIGDO_OPTS $opts $dirs CD$n"
+        echo $CMD
+        echo $CMD > CD$n/.disk/mkisofs
+        eval "$CMD"
 
     else
         echo "Neither jigdo nor iso wanted for CD$n"
